### PR TITLE
Move SlobrokMonitor to Java

### PR DESCRIPTION
**Pull request by hakonhall**

### CODE REVIEW

File: service-monitor/src/main/java/com/yahoo/vespa/service/monitor/SlobrokMonitor.java [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+        slobrokList.setup(slobroks.toArray(new String[0]));
+    }
+
+    Map<SlobrokServiceName, SlobrokServiceSpec> getRegisteredServices() {
+        if (!mirror.ready()) {
+            return new HashMap<>();
+        }
+
+        Mirror.Entry[] mirrorEntries = mirror.lookup("**");
+        return Arrays.asList(mirrorEntries).stream().collect(Collectors.toMap(
```
Comment (bjorncs) [inline service-monitor/src/main/java/com/yahoo/vespa/service/monitor/SlobrokMonitor.java]: Use `Arrays.stream()`

**Review by bjorncs — CHANGES_REQUESTED** [review]
Overall looking good. See comments.

**Comment by bjorncs** [comment]
Note that the default access modifier of Scala is `public`.

File: service-monitor/src/main/java/com/yahoo/vespa/service/monitor/SlobrokMonitor.java [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+import com.yahoo.jrt.slobrok.api.Mirror;
+import com.yahoo.jrt.slobrok.api.SlobrokList;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class SlobrokMonitor {
```
[the remark this answers is not in the document]
Comment (hakonhall) [reply service-monitor/src/main/java/com/yahoo/vespa/service/monitor/SlobrokMonitor.java]: Fixed

File: service-monitor/src/main/java/com/yahoo/vespa/service/monitor/SlobrokMonitor.java [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+    private final SlobrokList slobrokList = new SlobrokList();
+    private final Mirror mirror = new Mirror(supervisor, slobrokList);
+
+    void setSlobrokConnectionSpecs(List<String> slobroks) {
+        slobrokList.setup(slobroks.toArray(new String[0]));
+    }
+
+    Map<SlobrokServiceName, SlobrokServiceSpec> getRegisteredServices() {
+        if (!mirror.ready()) {
+            return new HashMap<>();
```
[the remark this answers is not in the document]
Comment (hakonhall) [reply service-monitor/src/main/java/com/yahoo/vespa/service/monitor/SlobrokMonitor.java]: Fixed

File: service-monitor/src/main/java/com/yahoo/vespa/service/monitor/SlobrokMonitor.java (hunk 1 above)
Comment (hakonhall) [reply service-monitor/src/main/java/com/yahoo/vespa/service/monitor/SlobrokMonitor.java]: Fixed

File: service-monitor/src/main/java/com/yahoo/vespa/service/monitor/SlobrokMonitor.java [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+    }
+
+    void shutdown() {
+        mirror.shutdown();
+    }
+
+    public static class SlobrokServiceName {
+        private final String name;
+
+        SlobrokServiceName(String name) {
```
[the remark this answers is not in the document]
Comment (hakonhall) [reply service-monitor/src/main/java/com/yahoo/vespa/service/monitor/SlobrokMonitor.java]: Yes, I prefer to limit access. Removed a bunch of other 'public' modifiers too.

**Comment by hakonhall** [comment]
@bjorncs PTAL

**Review by bjorncs — APPROVED** [review]
👍

**Pull request merged**